iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/CatalogUtils.java

```diff
@@ -0,0 +1,191 @@
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    String keyPrefix = CATALOG_CONFIG_PREFIX + catalogName;
+    conf.forEach(config -> {
+      if (config.getKey().startsWith(CatalogUtils.CATALOG_DEFAULT_CONFIG_PREFIX)) {
+        catalogProperties.putIfAbsent(
+            config.getKey().substring(CatalogUtils.CATALOG_DEFAULT_CONFIG_PREFIX.length()),
+            config.getValue());
+      } else if (config.getKey().startsWith(keyPrefix)) {
+        catalogProperties.put(
+            config.getKey().substring(keyPrefix.length() + 1),
+            config.getValue());
+      }
+    });
+
+    return catalogProperties;
+  }
+
+  public static String getCatalogName(Configuration conf) {
+    return MetastoreConf.getVar(conf, MetastoreConf.ConfVars.CATALOG_DEFAULT);
+  }
+
+  public static String getCatalogType(Configuration conf) {
+    return getCatalogType(conf, CatalogUtils.getCatalogName(conf));
+  }
+
+  public static String getCatalogType(Configuration conf, Properties catalogProperties) {
+    String catalogName = catalogProperties.getProperty(
+        CatalogUtils.CATALOG_NAME,
+        MetastoreConf.getVar(conf, MetastoreConf.ConfVars.CATALOG_DEFAULT));
+    return getCatalogType(conf, catalogName);
+  }
+
+  /**
+   * Get Hadoop config key of a catalog property based on catalog name
+   * @param catalogName catalog name
+   * @param catalogProperty catalog property, can be any custom property,
+   *                        a commonly used list of properties can be found
+   *                        at {@link org.apache.iceberg.CatalogProperties}
+   * @return Hadoop config key of a catalog property for the catalog name
+   */
+  public static String catalogPropertyConfigKey(String catalogName, String catalogProperty) {
+    return String.format("%s%s.%s", CATALOG_CONFIG_PREFIX, catalogName, catalogProperty);
+  }
+
+  /**
+   * Return the catalog type based on the catalog name.
+   * <p>
+   * See Catalogs documentation for catalog type resolution strategy.
+   *
+   * @param conf global hive configuration
+   * @param catalogName name of the catalog
+   * @return type of the catalog, can be null
+   */
+  public static String getCatalogType(Configuration conf, String catalogName) {
+    if (catalogName != null) {
+      String catalogType = conf.get(catalogPropertyConfigKey(
+          catalogName, CatalogUtil.ICEBERG_CATALOG_TYPE));
+      if (catalogName.equals(ICEBERG_HADOOP_TABLE_NAME)) {
+        return NO_CATALOG_TYPE;
+      } else {
+        return catalogType;
+      }
+    } else {
+      String catalogType = conf.get(CatalogUtil.ICEBERG_CATALOG_TYPE);
+      if (catalogType != null && catalogType.equals(LOCATION)) {
+        return NO_CATALOG_TYPE;
+      } else {
+        return catalogType;
+      }
+    }
+  }
+
+  public static String getCatalogImpl(Configuration conf, String catalogName) {
+    return Optional.ofNullable(catalogName)
```

Comment:
1. not sure if Optional makes code more readable here, but it's more like style preference
2. in `getCatalogType()` you only check `catalogName` for null. Should we check for an empty string as well?

Reply:
Added check for an empty string as well in `getCatalogType()`